rdkit/Chem/SaltRemover.py

```diff
@@ -33,20 +33,61 @@
 
 import os
 import re
+from rdkit.Chem.rdmolfiles import SDMolSupplier
 
 from rdkit import Chem
 from rdkit import RDConfig
 from contextlib import closing
 
+class Format:
+  Smarts, Mol, Smiles = range(3)
+
+class SmartsMolSupplier(object):
+  whitespace = re.compile(r'[\t ]+')
+
+  def __init__(self, defnFilename):
+    self.salts = []
+    inF = open(defnFilename, 'r')
+    with closing(inF):
+      for line in inF:
+        line = line.strip().split('//')[0]
+        if line:
+          salt = self.molFromSmartsLine(line)
+          if salt is None:
+            raise ValueError(line)
+          self.salts.append(salt)
+
+  @classmethod
+  def molFromSmartsLine(cls, line):
+    splitL = cls.whitespace.split(line)
+    return Chem.MolFromSmarts(splitL[0])
+
+  def __iter__(self):
+    return self
+
+  def __len__(self):
+    return len(self.salts)
+
+  def __getitem__(self, index):
+    return self.salts[index]
+
+  def next(self):
+    if self.current >= len(self.salts):
+      raise StopIteration
+    else:
+      salt = self.salts[self.current]
+      self.current += 1
+      return salt
 
 class SaltRemover(object):
   defnFilename = os.path.join(RDConfig.RDDataDir, 'Salts.txt')
 
-  def __init__(self, defnFilename=None, defnData=None):
+  def __init__(self, defnFilename=None, defnData=None, format=Format.Smarts):
```

Comment:
While it is possible to do, it is bad practice to use names of Python built-ins as variable names = `format` .

Use something like `fileFormat`, `dataFormat` or `defnFormat`